crates/e2e/tests/e2e/colocation_partial_fill.rs

```diff
@@ -99,6 +99,26 @@ async fn test(web3: Web3) {
             .contains(&buy_balance.as_u128())
     );
 
-    // TODO: test that we have other important per-auction data that should have
-    // made its way into the DB.
+    onchain.mint_blocks_past_reorg_threshold().await;
+
+    let settlement_event_processed = || async {
+        let order = services.get_order(&uid).await.unwrap();
+        if let OrderClass::Limit(LimitOrderClass {
+            executed_surplus_fee,
+        }) = order.metadata.class
+        {
+            executed_surplus_fee > U256::zero()
+        } else {
+            panic!("order is not a limit order");
+        }
+    };
+    wait_for_condition(TIMEOUT, settlement_event_processed)
+        .await
+        .unwrap();
+
+    let tx_hash = services.get_trades(&uid).await.unwrap()[0].tx_hash.unwrap();
+    let competition = services.get_solver_competition(tx_hash).await.unwrap();
+    assert!(!competition.common.solutions.is_empty());
+    assert!(competition.common.auction.orders.contains(&uid));
+    assert!(competition.common.auction.orders.contains(&uid));
```

Comment:
This check is duplicated. Can it be deleted or was this a copy-paste error and there was something else you wanted to check?